src/pages/AddEditCustomer.tsx

```diff
@@ -0,0 +1,745 @@
+    isLoading: isCustomerLoading,
+    isError: isCustomerError,
+  } = useGetCustomerByIdQuery(customerId || '0', { skip: !isEditMode });
+  const [createCustomer, { isLoading: isCreateCustomerLoading }] = useCreateCustomerMutation();
+  const [updateCustomerById, { isLoading: isUpdateCustomerLoading }] = useUpdateCustomerByIdMutation();
+  const [errorMessages, setErrorMessages] = useState([]);
+  const navigate = useNavigate();
+
+  const schema = yup
+    .object({
+      additional_information: yup.string().nullable(),
+      has_children: yup.boolean().nullable(),
+      has_hitas_ownership: yup.boolean().nullable(),
+      has_secondary_profile: yup.boolean(),
+      is_age_over_55: yup.boolean().nullable(),
+      is_right_of_occupancy_housing_changer: yup.boolean().nullable(),
+      last_contact_date: yup
+        .string()
+        .nullable()
+        .transform((val, originalVal) => (originalVal === '' ? null : val)),
+      right_of_residence: yup
+        .number()
+        .nullable()
+        .transform((val, originalVal) => (originalVal === '' ? null : val)),
+      primary_profile: yup
+        .object({
+          city: yup.string().required(t(`${T_PATH}.required`)),
+          contact_language: yup
+            .string()
+            .oneOf(langCodes)
+            .required(t(`${T_PATH}.required`)),
+          date_of_birth: yup.string().required(t(`${T_PATH}.required`)),
+          email: yup
+            .string()
+            .email()
+            .required(t(`${T_PATH}.required`)),
+          first_name: yup.string().required(t(`${T_PATH}.required`)),
+          last_name: yup.string().required(t(`${T_PATH}.required`)),
+          national_identification_number: yup.string().nullable(),
+          phone_number: yup.string().required(t(`${T_PATH}.required`)),
+          postal_code: yup.string().required(t(`${T_PATH}.required`)),
+          street_address: yup.string().required(t(`${T_PATH}.required`)),
+        })
+        .required(),
+      secondary_profile: yup
+        .object()
+        .when('has_secondary_profile', {
+          is: true,
+          then: yup.object({
+            city: yup.string().required(t(`${T_PATH}.required`)),
+            contact_language: yup
+              .string()
+              .oneOf(langCodes)
+              .required(t(`${T_PATH}.required`)),
+            date_of_birth: yup.string().required(t(`${T_PATH}.required`)),
+            email: yup
+              .string()
+              .email()
+              .required(t(`${T_PATH}.required`)),
+            first_name: yup.string().required(t(`${T_PATH}.required`)),
+            last_name: yup.string().required(t(`${T_PATH}.required`)),
+            national_identification_number: yup.string().nullable(),
+            phone_number: yup.string().required(t(`${T_PATH}.required`)),
+            postal_code: yup.string().required(t(`${T_PATH}.required`)),
+            street_address: yup.string().required(t(`${T_PATH}.required`)),
```

Comment:
```yup.string().required(t(`${T_PATH}.required`))``` seems to be used a lot here. Maybe something like
```
const requiredString = () => yup.string().required(t(`${T_PATH}.required`));

...

street_address: requiredString(),
```
could make sense?

Reply:
Makes total sense. There weren't that many in the beginning, until I noticed that the backend kinda requires all of the fields 😅